lms/djangoapps/discussion_api/api.py

```diff
@@ -361,21 +366,52 @@ def get_comment_list(request, thread_id, endorsed, page, page_size):
     return get_paginated_data(request, results, page, num_pages)
 
 
-def _check_editable_fields(cc_content, data, context):
+def _check_fields(allowed_fields, data, message):
     """
-    Raise ValidationError if the given update data contains a field that is not
-    in editable_fields.
+    Raise ValidationError if the given data contains a key that is not in
+    allowed_fields
     """
-    editable_fields = get_editable_fields(cc_content, context)
-    non_editable_errors = {
-        field: ["This field is not editable."]
-        for field in data.keys()
-        if field not in editable_fields
-    }
+    non_editable_errors = {field: [message] for field in data.keys() if field not in allowed_fields}
```

Comment:
nit: since the function name/logic has been generalized to "allowed" instead of "editable", please rename this variable accordingly